File: Tutorial5-3Dagger-Android-MutipleActivitiesAndFragments/src/main/java/com/example/tutorial5_3dagger_android_mutipleactivitiesandfragments/model/DummyDependency.java
package com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.model;

import android.content.Context;

import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.R;


public class DummyDependency {

    private Context context;

    public DummyDependency(Context context) {
        this.context = context;
    }

    public String getApplicationName() {
        return context.getString(R.string.app_name);
    }

}

File: Tutorial5-3Dagger-Android-MultipleActivitiesAndFragments/src/main/java/com/example/tutorial5_3dagger_android_multipleactivitiesandfragments/model/DummyDependency.kt
package com.example.tutorial5_3dagger_android_multipleactivitiesandfragments.model

import android.content.Context
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.R

class DummyDependency(private val context: Context) {

    val applicationName: String
        get() = context.getString(R.string.app_name)

}